pkg/workflows/upgrade_test.go

```diff
@@ -114,9 +119,11 @@ func newUpgradeManagedClusterTest(t *testing.T) *upgradeTestSetup {
 	return tt
 }
 
-func (c *upgradeTestSetup) expectSetup() {
+func (c *upgradeTestSetup) expectSetup(expectedCluster *types.Cluster) {
```

Comment:
why do we need to pass expected cluster? isn't it always the same one? `tt.managementCluster`

Reply:
No it changes for unit tests that upgrade additional workload clusters

Reply:
Just for my understanding, could you describe how and when it changes?
Happy to take this offline if it's too long :)

Reply:
Of course! Since I moved the code to get the current cluster spec to the `setup-and-validate` task, I had to move the `EXPECT()` for this method here in the unit tests. And when we are upgrading the workload clusters, we pass in the management cluster as the expected cluster here (for the other tests, we pass in the workload cluster). So it's the same logic that was in the test before, just moved to the setup task.